pkg/kubectl/cmd/logs.go

```diff
@@ -162,18 +175,34 @@ func (o *LogsOptions) Complete(f *cmdutil.Factory, out io.Writer, cmd *cobra.Com
 	o.ClientMapper = resource.ClientMapperFunc(f.ClientForMapping)
 	o.Out = out
 
+	if selector != "" {
+		if logOptions.Follow {
+			return cmdutil.UsageError(cmd, "only one of follow (-f) or selector (-l) is allowed")
+		}
+		if logOptions.Container != "" {
+			return cmdutil.UsageError(cmd, "a container cannot be specified when using a selector (-l)")
+		}
+		if logOptions.TailLines == nil {
+			logOptions.TailLines = &selectorTail
```

Comment:
Why using a selector would give me different behavior than without one? By default this is not set, iow. it's unlimited. I'd like this to be consistent.